File: leetcode.com/0219_contains_duplicate_2/solution.go
func containsNearbyDuplicate(nums []int, k int) bool {
    h := make(map[int]int)

    for i, n := range(nums) {

        if v, ok := h[n]; ok {
            if i - v <= k {
                return true
            }
        }

        h[n] = i
    }

    return false
}

File: leetcode.com/0219_contains_duplicate_2/solution.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        d = collections.defaultdict(int)

        for i, n in enumerate(nums):
            if n in d and abs(d[n] - i) <= k:
                return True

            d[n] = i

        return False
